sdk/core/core-http/test/policies/atomSerializationPolicyTests.ts

```diff
@@ -0,0 +1,162 @@
+
+    const indexOfOpenUpdateTag = request.body.indexOf("<updated>");
+    const indexOfCloseUpdateTag = request.body.indexOf("</updated>");
+    assert.equal(
+      request.body.substring(0, indexOfOpenUpdateTag),
+      expectedRequestBody.substring(0, indexOfOpenUpdateTag),
+      "Atom XML serialization failure"
+    );
+    assert.equal(
+      request.body.substring(indexOfCloseUpdateTag),
+      expectedRequestBody.substring(indexOfCloseUpdateTag),
+      "Atom XML serialization failure"
+    );
+  });
+
+  it("should deserialize properly with xml response body and given AtomXMLOperationSpec", async function() {
+    const request: WebResource = createRequest({
+      serializer: new TestSerializer()
+    });
+
+    const mockClient: HttpClient = {
+      sendRequest: (req) =>
+        Promise.resolve({
+          request: req,
+          status: 200,
+          headers: new HttpHeaders({
+            "content-type": "application/xml"
+          }),
+          bodyAsText: `<entry xmlns="http://www.w3.org/2005/Atom"><id>https://dummy.servicebus.windows.net/dummy?api-version=2017-04</id><title type="text">dummy</title><author><name>dummy</name></author><content type="application/xml"><QueueDescription xmlns="http://schemas.microsoft.com/netservices/2010/10/servicebus/connect" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><LockDuration>PT2M</LockDuration><MaxSizeInMegabytes>1024</MaxSizeInMegabytes></QueueDescription></content></entry>`
+        })
+    };
+
+    const policy = atomSerializationPolicy().create(mockClient, new RequestPolicyOptions());
+    await policy.sendRequest(request);
+  });
+});
+
+function createRequest(atomXmlOperationSpec?: AtomXmlOperationSpec): WebResource {
+  const request = new WebResource();
+  request.atomXmlOperationSpec = atomXmlOperationSpec;
+  return request;
+}
+
+class TestSerializer implements AtomXmlSerializer {
+  serialize(resource: any): object {
+    const property1 = "LockDuration";
+    const property2 = "MaxSizeInMegabytes";
+
+    const resourceName = "QueueDescription";
+    const content: any = {};
+    content[resourceName] = {
+      $: {
+        xmlns: "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect"
+      }
+    };
+
+    content[resourceName][property1] = resource[property1];
+    content[resourceName][property2] = resource[property2];
+    content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+    return {
+      entry: {
+        $: {
+          xmlns: "http://www.w3.org/2005/Atom"
+        },
+        updated: new Date().toISOString(),
+        content: content
+      }
+    };
+  }
+
+  async deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse> {
+    const expectedParsedBody = {
```

Comment:
I think it'd be nice to have the expectedParsedBody as something you pass in to the `TestSerializer` constructor if possible. You could instantiate it once and use it for all the tests, but that will make it possible to test parsing other bodies. I also didn't know this assertion was even being performed in the tests until I made it to this point.

Reply:
Sure! This was added because changes were blocked on earlier feedback to have this coverage exist alongside the policy tests (so that we were hitting the XML utilities code paths). 
Ideally we should fix missing unit test coverage for the XML utilities, that way we don't need to have test serializers and revert to using mock serializer. Filed issue for it at - https://github.com/Azure/azure-sdk-for-js/issues/5548